R/fit_peak.R

```diff
@@ -0,0 +1,173 @@
+#' `observation`, which represents the data points, and `weight`, which is the
+#' importance assigned to an observation. Higher weights indicate that an
+#' observation has more influence on the model outcome, while lower weights
+#' reduce its impact. The structure is:
+#' `weighted_observations = tibble(observation = c(...), weight = c(...))`
+#' @param conf_levels The confidence levels for parameter estimates, a numeric
+#' vector of length 3. Default is c(0.50, 0.90, 0.95).
+#' @param family A character string specifying the family for
+#' modeling. Choose between "weibull", "lnorm" or "exp".
+#' @param optim_method A character string specifying the method to be used in
+#' the optimisation. Lookup ?optim::stats for details about methods.
+#' Default is Nelder-Mead. If using the exp family it is recomended to use
+#' Brent as it is a one-dimensional optimisation.
+#' @param lower_optim A numeric value for the optimisation. Default is -Inf.
+#' @param upper_optim A numeric value for the optimisation. Default is Inf.
+#'
+#' @return A tibble containing:
+#'   - 'high_level': The highest peak intensity level
+#'   - 'medium_level': The medium peak intensity level
+#'   - 'low_level': The lowest intensity level
+#'   - 'optim_fit_par_1': The first fit parameter for the chosen family.
+#'       - For 'weibull': Shape parameter (k).
+#'       - For 'lnorm': Mean of the log-transformed observations.
+#'       - For 'exp': Rate parameter (λ).
+#'   - 'optim_fit_par_2': The second fit parameter for the chosen family.
+#'      - For 'weibull': Scale parameter (λ).
+#'      - For 'lnorm': Standard deviation of the log-transformed observations.
+#'      - For 'exp': Not applicable (set to NA).
+#'   - 'obj_value': The value of the objective function
+#'     (negative log-likelihood), which represent the minimized objective
+#'      function value from the optimisation. Smaller value equals better
+#'      optimisation.
+#'   - 'converged': Logical. TRUE if the optimisation converged.
+#'   - 'family': The distribution family used for the optimization.
+#'      - 'weibull': Uses the Weibull distribution for fitting.
+#'      - 'lnorm': Uses the Log-normal distribution for fitting.
+#'      - 'exp': Uses the Exponential distribution for fitting.
+#'
+#' @export
+#'
+#' @examples
+#' # Create three seasons with random observations
+#' obs = 10
+#' season = c("2018/2019", "2019/2020", "2020/2021")
+#' season_num_rev <- rev(seq(from = 1, to = length(season)))
+#' observations = rep(stats::rnorm(10,obs), length(season))
+#'
+#' # Add into a tibble with weight decreasing going one season step back
+#' peak_input <- tibble::tibble(
+#'   observation = observations,
+#'   weight = 0.8^rep(season_num_rev, each = obs)
+#' )
+#'
+#' # Use the peak model
+#' fit_peak(weighted_observations = peak_input,
+#'          conf_levels = c(0.50, 0.90, 0.95),
+#'          family = "weibull")
+
+fit_peak <- function(
```

Comment:
Is your function actually fitting a peak? 
It seems more like you are predicting next peak?
Maybe call it "predict_next_peak"?